core/src/main/scala/org/apache/spark/status/AppStatusListener.scala

```diff
@@ -645,8 +645,11 @@ private[spark] class AppStatusListener(
   }
 
   override def onTaskEnd(event: SparkListenerTaskEnd): Unit = {
-    // TODO: can this really happen?
-    if (event.taskInfo == null) {
+    // TODO: can taskInfo null really happen?
+    // For resubmitted tasks caused by ExecutorLost, the SparkListenerTaskEnd is useless and
+    // will make activeTask in stage to be negative, this will cause stage not be removed in
+    // liveStages, and finally cause executor not removed in deadExecutors
```

Comment:
```suggestion
    // SPARK-41187: For `SparkListenerTaskEnd` with `Resubmitted` reason, which is raised by executor lost,
    // it can lead to negative `LiveStage.activeTasks` since there's no corresponding `SparkListenerTaskStart`
    // event for each of them. The negative activeTasks will make the stage always remains in the live stage list
    // as it can never meet the condition activeTasks == 0. This in turn causes the dead executor to never be
    // retained if that live stage's submissionTime is less than the dead executor's removeTime( see
    // isExecutorActiveForLiveStages). Since this kind of `SparkListenerTaskEnd` is useless here, we simply
    // ignore it.
```

Reply:
> 

You make it more clearly, great